### DISCUSSION TITLE
Move to the backlog - Transparency - community needs fixes to move to maui.

### DISCUSSION THREAD

**Discussion by developer9969** [dotnet/maui#14263] [General]
**Let me start total admiration for the team and what they achieve considering how small they are as a team..**

@davidortinau @maddymontaquila 
Unfortunately you are the face of Maui and I bet you have your hands tied too but the state of Maui continue to be a problem

We are trying very very hard to convert applications written in Xamarin to MAUI but we face a daily struggle to get things working.

We are stuffing our apps with workarounds everywhere and something that should be just a port has quickly turned into a massive scope creep because simple things just dont work on Maui.. I know you know this but we need some transparency and some quick action from  yourself - We love the initiative that @PureWeen  took with his nuget - it cleared quite a few bugs for us.. why dont you do the same for collectionview??

Basic things that should just work and they dont 

- CollectionView - resize - EmptyView  and a miriad of bugs but EmptyView not working is a  pain, and involves a big workaround for us with crappy code to hide /displays screens.
- Gesture recognizers  - Common !!  we had to go on discovery tour to find out which control they work on - they dont work on collectionview and frame etc...  
- Line breaks in entry -  here we go again writing handlers to fix this.
- Release mode has some nasties - I.E Visual state on iOS does not work
- Image -clipping with borders do not work
- Keyboard issues on iOS covering the entry.
- We have no official UI Testing framework

The list goes on - all of these items have already been raised as bugs  but never fixed - we need the basic stuff working!!!!

We seem to get the dreaded "Moved to the backlog"  we all have one of those and wherever I worked means - "It will never be fixed!!"

I am trying to be constructive here , but unless you start dogfooding - writing apps that go live - you will never hit a miriad of issues that we encounter when we are writing real apps...

I know you are forced to move to the .net 8 bandwagon , but here we are in a very much broken state till Nov 2023 with the hope that magically it will all start working... 

We totally understand that any framework has bugs but it defeats any logic in trying to even support windows when you can barely support iOS and Android.

@maddymontaquila   HotReload - we desperatly need this working reliably , its a massive hit and miss -  
Ask one of your developers to build some apps - with 10 -20 screens - popups - bottomsheet- collectionViews etc and then try to use hot reload in both ios and android - than you will know what we are talking about!

I hope I get some response because I am struggling to justify to managment on why its taking me too long to port the app to Maui!!

Finally -Why cant you give us an interim nuget with fixes like @PureWeen  does ? that would help the community enormosly!
Its not ideal but its a solution that will give us a chance to move on into the Maui world..

thats all

**Comment by jsuarezruiz** [comment]
First of all, thanks for the detailed feedback, I have shared it with PMs and I am sure they will follow/reply with more details. My answer is more to share that we are trying to improve the process of prioritizing issues, and fixes and also the transparency/communication about the status of everything. About the mentioned issues, there are several in progress or recently merged.

- CollectionView - resize - EmptyView: https://github.com/dotnet/maui/pull/11763
- Gesture recognizers: This https://github.com/dotnet/maui/pull/12259 should impact in some of the controls where are not working and there are other fixes like https://github.com/dotnet/maui/pull/12027 in progress.
- Image -clipping: Should be fixed by https://github.com/dotnet/maui/pull/10964
- Keyboard issues on iOS covering the entry: Should have been fixed by https://github.com/dotnet/maui/pull/13499 and https://github.com/dotnet/maui/pull/14012

And there are still important pending things from your list, we are pushing to have it as soon as possible.

On the other hand, some examples of Xamarin.Forms have been ported and also create periodic UI Challenges like these:

- https://github.com/jsuarezruiz/netmaui-movies-app-challenge
- https://github.com/jsuarezruiz/netmaui-skateboard-ecommerce-app-challenge
- https://github.com/jsuarezruiz/netmaui-finance-app-challenge
- https://github.com/jsuarezruiz/netmaui-moon-nft-app-challenge
- https://github.com/jsuarezruiz/netmaui-coupons-app-challenge
- https://github.com/jsuarezruiz/netmaui-surfing-app-challenge
- https://github.com/jsuarezruiz/netmaui-beautyshop-app-challenge
- Etc

With the aim of doing dogfooding, having the experience of a developer, detecting what is going well and what is going wrong, [filling in issues](https://github.com/jsuarezruiz/netmaui-surfing-app-challenge/issues/1), etc.

**Comment by prok155** [comment]
I'm trying to migrate my app from Xamarin to MAUI from May 2022 until now and every time I try to do that, I face new bugs which don't allow to migrate my app. Of course, I open new issues on github, but they are always moved to backlog and it takes several months to fix it.

Xamarin support ends in May 2024 and I guess that after that time it will be impossible to do updates in app stores and google store, because both stores requires the latest sdk. I'm afraid, because I have apps in both stores and it doesn't seem that issues will be solved faster. 


I think that the problem is that the MAUI project is too big and the team is to small.

**Comment by developer9969** [comment]
@jsuarezruiz  thank you for answering!!  hey just to say whenever I need some inspiration  I always look at your work or the pages you maintain - 

its tough here ... and those fixes you mentioned they would help a great deal. Will they be available in next Visual Studio release?  the gestures - emptyView - keyboard - those are just blockers for us as they are the backbone of any applications with data.

Regarding dogfooding though - those are just samples with  no external pressure if things do not work.  If you add  - outlook - or teams or whatever that is used by thousands of users and suddendly something basic does not work, the pressure to fix it would be right there!! do you see what I mean ? 

the gesture issue https://github.com/dotnet/maui/issues/8895

Visual state - 
In our apps Visual state plays a massive role  and everywhere (77 pages) buttons are enabled depending on entries etc...
all fine in android but iOS Release mode just does not work

https://github.com/dotnet/maui/issues/11662

All the above are basic fix - that have an impact on any app -

**Comment by davidortinau** [comment]
"Moved to the backlog" here means it's not being worked in the current sprint. We plan sprint by sprint. You can look at the release notes and you'll see the issues fixed probably all spent some time in the backlog state. 

We have a .NET 7 service release coming in a few weeks and with few exceptions ship monthly. You can preview the release notes right now https://github.com/dotnet/maui/releases/tag/untagged-0d3d534408ebca4e2147.

.NET 8 introduces NuGet packages you can download directly from public build pipelines. That is the equivalent of the patch package you're asking for, though it does mean adopting preview releases. You can also grab code right from the PRs and add them to your project code directly when waiting for a public release isn't acceptable.

If you'd like to have a call and go deeper on the issues blocking your upgrade, then send me an email and I'll set that up. david.ortinau@microsoft.com

**Comment by akhanalcs** [comment]
> We have no official UI Testing framework

Any new update on this?

**Comment by developer9969** [comment]
@davidortinau  thank you for replying .

**Backlog**
We work in sprint too - but allow me saying this, there are so many bugs  sitting in the backlog for months  that quite important - you could argue that what is important to me is not important to the Maui team or others and you have a difficult job to decide what should be done..
but let me say this -  Xamarin was not perfect and was a rollercoster too but is stable... With Maui when you write an app not a single page you will encounter so many issues that makes so hard to port it.

**We need at least the basic working** 

- Collectionview
- Gestures
- Borders  ( we no longer use frame) 
- Visual state ( https://github.com/dotnet/maui/issues/11662)
- Hot reload (this is a top top priority) the dev loop is too long with this not working properly.

above are some of the cornerstone of any app, if those are buggy in basic areas we are truly doomed! CollectionView is a massive problem.

When you have 50-60 pages you use styles as its impossible otherwise, well we use visual state for everything and guess what buttons styles does not work in release mode only for iOS that causes a big issue with so many pages relying on it.

CollectionView Empty view ( you already have the fix ) can you merge it please https://github.com/dotnet/maui/pull/11763

A meeting is not necessary and I truly appreciate the offer, we just need the core controls actually working and more frequent releases,
a may be (unless already exists)  a suggestion "top 5 bugs that the community can vote for your team to fix in a sprint)

there is a feeling in the community that for reasons that you know things are not being fixed but just pushed back

I have not told you anything new but hopefully you can help give us something (fixing the core controls) that will help us porting the apps to Maui.
